ProcessLib/InitialCondition.cpp

```diff
@@ -23,17 +23,42 @@
 namespace ProcessLib
 {
 std::unique_ptr<InitialCondition> createUniformInitialCondition(
-    BaseLib::ConfigTree const& config, int const /*n_components*/)
+    BaseLib::ConfigTree const& config, int const n_components)
 {
     //! \ogs_file_param{initial_condition__type}
     config.checkConfigParameter("type", "Uniform");
 
-    //! \ogs_file_param{initial_condition__Uniform__value}
-    auto value = config.getConfigParameter<double>("value");
-    DBUG("Using value %g", value);
+    if (n_components == 1)
+    {
+        //! \ogs_file_param{initial_condition__Uniform__value}
+        auto const value = config.getConfigParameterOptional<double>("value");
+        if (value)
+        {
+            DBUG("Using value %g for the initial condition.", *value);
+            return std::unique_ptr<InitialCondition>{
+                new UniformInitialCondition{std::vector<double>{*value}}};
+        }
```

Comment:
Yes. Adding clarifying comment in the code.
Idea is that 'value' is optional for single-component variables and in that case if we find 'value' in config it is OK, else the 'value**s**' are required.